### PR TITLE
Netflow CRD: handle empty attributes checks

### PR DESCRIPTION
**Pull request by tanyatukade**
- Added checks to handle empty attributes. If empty optional attribute detected, it sets to default values defined by apic docs.
- Added few more test cases in test file.
- Changed attribute name "type" (version) to "flowType" to avoid confusion with yaml native type.

### CODE REVIEW

File: pkg/controller/netflow.go [hunk 1]
```diff
@@ -150,23 +150,41 @@ func (cont *AciController) handleNetflowPolUpdate(obj interface{}) bool {
...
 	VmmVSwitch := apicapi.NewVmmVSwitchPolicyCont(cont.vmmDomainProvider(), cont.config.AciVmmDomain)
 	RsVmmVSwitch := apicapi.NewVmmRsVswitchExporterPol(cont.vmmDomainProvider(), cont.config.AciVmmDomain, nfDn)
 	VmmVSwitch.AddChild(RsVmmVSwitch)
-	RsVmmVSwitch.SetAttr("activeFlowTimeOut", strconv.Itoa(nfp.Spec.FlowSamplingPolicy.ActiveFlowTimeOut))
-	RsVmmVSwitch.SetAttr("idleFlowTimeOut", strconv.Itoa(nfp.Spec.FlowSamplingPolicy.IdleFlowTimeOut))
-	RsVmmVSwitch.SetAttr("samplingRate", strconv.Itoa(nfp.Spec.FlowSamplingPolicy.SamplingRate))
+	if nfp.Spec.FlowSamplingPolicy.ActiveFlowTimeOut != 0 {
+		RsVmmVSwitch.SetAttr("activeFlowTimeOut", strconv.Itoa(nfp.Spec.FlowSamplingPolicy.ActiveFlowTimeOut))
+	} else {
+		RsVmmVSwitch.SetAttr("activeFlowTimeOut", "60")
```
Comment (tomflynn) [inline pkg/controller/netflow.go:173]: Do you need the else for these properties? The default is already set to this value anyway in apic_metadata.go

File: pkg/controller/netflow_test.go [hunk 2]
```diff
@@ -124,24 +124,40 @@ func (cont *AciController) netflowPolUpdate(nfp *netflowpolicy.NetflowPolicy) ap
 	nf.SetAttr("dstAddr", nfp.Spec.FlowSamplingPolicy.DstAddr)
 	if nfp.Spec.FlowSamplingPolicy.Version == "netflow" {
 		nf.SetAttr("ver", "v5")
-	}
-	if nfp.Spec.FlowSamplingPolicy.Version == "ipfix" {
+	} else if nfp.Spec.FlowSamplingPolicy.Version == "ipfix" {
```
Comment (tomflynn) [inline pkg/controller/netflow_test.go:127]: I notice that this func is duplicating most of what handleNetflowPolUpdate does in netflow.go. Can you rework the code to exercise handleNetflowPolUpdate instead?

**Review by tomflynn — CHANGES_REQUESTED** [review]

File: pkg/controller/netflow.go (hunk 1 above)
Comment (tanyatukade) [reply pkg/controller/netflow.go:173]: The apic_metadata.go has the info for default values but the setAttr is the function that actually sets the attribute with the value.

File: pkg/controller/netflow_test.go (hunk 2 above)
Comment (tanyatukade) [reply pkg/controller/netflow_test.go:127]: Will work on that.

**Review by tomflynn — APPROVED** [review]

**Pull request merged**